src/HTSPTypes.cpp

```diff
@@ -437,3 +441,394 @@ void AutoRecording::SetFulltext(uint32_t fulltext)
+        }
+        else
+        {
+          /* Do not overwrite SUBSCRIPTION_FORCERUNNING_FAILED */
+          if (GetState() != SUBSCRIPTION_FORCERUNNING_FAILED)
+            SetState(SUBSCRIPTION_NOFREEADAPTER);
+        }
+      }
+      else
+        SetState(SUBSCRIPTION_UNKNOWN);
+
+      /* Show an OSD message */
+      ShowStateNotification();
+    }
+    else
+      SetState(SUBSCRIPTION_RUNNING);
+  }
+  else
+  {
+    /* This field is absent when everything is fine */
+    if (status != NULL)
+    {
+      XBMC->QueueNotification(ADDON::QUEUE_INFO, status);
+      SetState(SUBSCRIPTION_UNKNOWN);
+    }
+    else
+      SetState(SUBSCRIPTION_RUNNING);
+  }
+}
+
+void Subscription::ShowStateNotification(void)
+{
+  if (GetState() == SUBSCRIPTION_NOFREEADAPTER)
+    XBMC->QueueNotification(ADDON::QUEUE_WARNING, XBMC->GetLocalizedString(30450));
+  else if (GetState() == SUBSCRIPTION_SCRAMBLED)
+    XBMC->QueueNotification(ADDON::QUEUE_WARNING, XBMC->GetLocalizedString(30452));
+  else if (GetState() == SUBSCRIPTION_NOSIGNAL)
+    XBMC->QueueNotification(ADDON::QUEUE_WARNING, XBMC->GetLocalizedString(30453));
+  else if (GetState() == SUBSCRIPTION_UNKNOWN)
+    XBMC->QueueNotification(ADDON::QUEUE_ERROR,   XBMC->GetLocalizedString(30454));
+  else if (GetState() == SUBSCRIPTION_FORCERUNNING_FAILED)
+    XBMC->QueueNotification(ADDON::QUEUE_WARNING, XBMC->GetLocalizedString(30455));
+  else if (GetState() == SUBSCRIPTION_USERLIMIT)
+    XBMC->QueueNotification(ADDON::QUEUE_WARNING, XBMC->GetLocalizedString(30456));
+}
+
+/* NOTE: This code is running in a separate thread */
+void Subscription::ShowConflictDialog(void)
+{
+  tvhdebug("demux conflict dialog: open, state: %i, subscription id: %i", GetState(), GetId());
+
+  /* Save the initial subscription id */
+  uint32_t initialId = GetId();
+
+  /* Save the initial subscription state */
+  eSubsriptionState initialState = GetState();
+
+  /* As we saved the initial state, we can change this now */
+  SetState(SUBSCRIPTION_NOFREEADAPTER);
+
+  /*
+   * Conflict case 1: (initialState == SUBSCRIPTION_RUNNING)
+   * Subscription was running before, but the adapter got stolen by another subscription
+   * Ask user if he wants to continue watching by hijacking an adapter back (increase weight)
+   *
+   * Conflict case 2: (initialState == SUBSCRIPTION_STARTING)
+   * No free adapter found to start this channel from the beginning on
+   * Ask user if he wants to hijack an adapter (increase weight)   'DIALOG_NOSTART_DELAY' seconds of idling
+   * This time of idling is needed to prevent the dialog from popping up when zapping
+   */
+
+  if (initialState == SUBSCRIPTION_STARTING)
+  {
+    /*
+     * Show dialog only after 'DIALOG_NOSTART_DELAY' seconds with this state
+     * This to prevent the dialog from popping up when zapping
+     */
+    sleep(DIALOG_NOSTART_DELAY);
+
+    /* Break if the conflict was solved */
+    if (IsConflictSolved(initialId))
+      return;
+  }
+
+  GUI->Lock();
```

Comment:
See the subsequent lines, a dialog is shown asking for user input. The lock is released after that call. Are you saying no locking is required?